### PR TITLE
Codelabs index + cards

### PR DESCRIPTION
**Pull request by skarEE**
Contains basic layout
Creation of cards from ul, could be a ol?
Sorting

Updates #20 and #21

### CODE REVIEW

File: google-codelabs/google_codelabs.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+      this.sortBy_ = sortBy;
+      this.cards_ = /** @type {!Cards} */ (cards);
+    }
+
+    this.hasSetup_ = true;
+  }
+}
+
+window.customElements.define('google-codelabs', Codelabs);
```
Comment (x1ddos) [inline google-codelabs/google_codelabs.js]: As agreed in chat, rename this to `google-codelabs-index`?

**Comment by skarEE** [comment]
PTAL

File: google-codelab-index/google_codelab_index_cards.js [hunk 2]
```diff
@@ -164,13 +164,25 @@ class Cards extends HTMLElement {
...
 
+  /**
+   * 
+   * @param {string} category 
+   * @returns {string}
+   */
+  normalizeCategory_(category) {
+    return category.toLowerCase()
+        .replace(/\s+/g, '-')           // Replace spaces with -
+        .replace(/\-\-+/g, '-')         // Replace multiple - with single -
```
Comment (x1ddos) [inline google-codelab-index/google_codelab_index_cards.js]: This is weird. Normally, `-` outside of square brackets `[ ]` shouldn't count as a special regexp character. In fact, I tried in chrome js console. Both work:

image

What can I say. Weird javascript. :)

File: google-codelab-index/_categories.scss [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    .#{$name}-bg {
+      background-color: $color;
+    }
+
+    @if $icon {
+      .#{$name}-icon {
+        background-image: url(//codelabs.developers.google.com/images/icons/#{$icon});
+      }
+    } @else {
+      .#{$name}-icon {
```
Comment (x1ddos) [inline google-codelab-index/_categories.scss]: Or the if/else could be inside a single `.#{$name}-icon` but I don't really care. You probably had other reasons for this.
Ah, maybe SCSS doesn't allow `@if` statements within a CSS class definition.

File: google-codelab-index/google_codelab_index_cards.js (hunk 2 above)
Comment (skarEE) [reply google-codelab-index/google_codelab_index_cards.js]: I probably stuffed up :)

File: google-codelab-index/_categories.scss (hunk 3 above)
Comment (skarEE) [reply google-codelab-index/_categories.scss]: It totally can!

**Pull request merged**